Assets/Script/Gameplay/HUD/Practice/PracticeSectionView.cs

```diff
@@ -101,5 +105,157 @@ public void UpdateView()
+                return sectionName.Replace('_', ' ');
+            }
+
+            if (number != null)
+            {
+                return Localize.KeyFormat(("Gameplay", "Practice", "SectionFormats", "WithNumber"), localizedName, number);
+            }
+
+            return localizedName;
+        }
+
+        /** 
+         * Whether the string follows the convention for letter-based section names
+         */
+        private bool IsLetterBasedSectionName(string text)
+        {
+            if (text.Length == 0 || !text[0].IsAsciiLetterLower())
+            {
+                return false;
+            }
+
+            if (text.Length == 1)
+            {
+                // e.g. "a" for "A section"
+                return true;
+            }
+
+            if (text.Length == 2)
+            {
+                // e.g. "a1" for "A section 1"
+                return text[1].IsAsciiDigit();
+            }
+
+            if (text[1] == '_')
+            {
+                // e.g. "a_b" for "A section B"
+                return text.Length == 3 && text[2].IsAsciiLetterLower();
+            }
+
+            if (text[1..^1].All(c => c.IsAsciiDigit()))
+            {
+                if (text[^1].IsAsciiDigit())
+                {
+                    // e.g. "a123" for "A section 123"
+                    return true;
+                }
+
+                if (text[^1].IsAsciiLetterLower())
+                {
+                    // e.g. "a123b" for "A section 123B"
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private (string name, string number) DeriveNameAndNumber(string section)
+        {
+            if (section.Contains('_') && !section.EndsWith('_'))
+            {
+                var indexOfLastUnderscore = section.LastIndexOf('_');
+                var lastWord = section[(indexOfLastUnderscore + 1)..];
+                if (IsSectionNumber(lastWord))
+                {
+                    var name = section[0..indexOfLastUnderscore];
+
+                    string normalizedSectionName;
+                    return (alternateSectionNames.TryGetValue(name, out normalizedSectionName) ? normalizedSectionName : name, lastWord);
+                }
+            }
+
+            string normalizedName;
+            return (alternateSectionNames.TryGetValue(section, out normalizedName) ? normalizedName : section, null);
+        }
+
+        // Whether the string follows the convention for section numbers (a, 1, 1a)
+        private bool IsSectionNumber(string text)
+        {
+            if (text.Length == 0)
+            {
+                return false;
+            }
+
+            if (text.Length == 1)
+            {
+                return text[0].IsAsciiLetterLower() || text[0].IsAsciiDigit();
+            }
+
+            return text[..^2].All(c => c.IsAsciiDigit()) && (text[^1].IsAsciiLetterLower() || text[^1].IsAsciiDigit());
```

Comment:
Since this runs during gameplay, I suspect some will be unhappy with the use of LINQ here.

You could instead do this with a for loop, though it wouldn't be as concise.

```suggestion
            for (int i = 0;i < text.Length - 2; i++) 
            {
                if (!text[i].IsAsciiDigit()) {
                    return false;
                }
            }
            return text[^1].IsAsciiLetterLower() || text[^1].IsAsciiDigit();
```

(not tested, just typed into GitHub's terrible little text box)